contrib/golang/filters/http/source/golang_filter.cc

```diff
@@ -1018,6 +1018,52 @@ CAPIStatus Filter::getStringValue(int id, GoString* value_str) {
   return CAPIStatus::CAPIOK;
 }
 
+CAPIStatus Filter::getDynamicMetadata(const std::string& filter_name, GoSlice* buf_slice) {
+  Thread::LockGuard lock(mutex_);
+  if (has_destroyed_) {
+    ENVOY_LOG(debug, "golang filter has been destroyed");
+    return CAPIStatus::CAPIFilterIsDestroy;
+  }
+
+  auto& state = getProcessorState();
+  if (!state.isProcessingInGo()) {
+    ENVOY_LOG(debug, "golang filter is not processing Go");
+    return CAPIStatus::CAPINotInGo;
+  }
+
+  if (!state.isThreadSafe()) {
+    auto weak_ptr = weak_from_this();
+    ENVOY_LOG(debug, "golang filter getDynamicMetadata posting request to dispatcher");
+    state.getDispatcher().post([this, &state, weak_ptr, filter_name, buf_slice] {
```

Comment:
use a `string::string&` might not safe here?

Reply:
IMO, `filter_name` will be copyed to the closure, so it is safe?

Reply:
I'm not sure about it, I know that `std::string&` is a pointer refer in normal function call case.
@wbpcode Could you please give some suggestions here? thanks very much.

Reply:
by default the lambda will copy the captured variable. The the filter_name will be copied.

Reply:
Great, thanks @wbpcode